uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-widget/leave-widget-next-leave.component.spec.js

```diff
@@ -0,0 +1,178 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'mocks/helpers/controller-on-changes',
+  'mocks/data/option-group-mock-data',
+  'mocks/apis/leave-request-api-mock',
+  'leave-absences/shared/components/leave-widget/leave-widget-next-leave.component'
+], function (_, moment, controllerOnChanges, OptionGroupData) {
+  describe('leaveWidgetNextLeave', function () {
+    var $componentController, $provide, $q, $rootScope, $scope, ctrl,
+      LeaveRequest, leaveRequestStatuses, OptionGroup, sharedSettings;
+    var childComponentName = 'leave-widget-next-leave';
+    var contactId = 101;
+
+    beforeEach(module('leave-absences.components.leave-widget',
+      'leave-absences.mocks', function (_$provide_) {
+        $provide = _$provide_;
+      }));
+
+    beforeEach(inject(function (LeaveRequestAPIMock, OptionGroupAPIMock) {
+      $provide.value('LeaveRequestAPI', LeaveRequestAPIMock);
+      $provide.value('OptionGroup', OptionGroupAPIMock);
+    }));
+
+    beforeEach(inject(['$componentController', '$q', '$rootScope',
+      'LeaveRequest', 'OptionGroup', 'shared-settings',
+      function (_$componentController_, _$q_, _$rootScope_, _LeaveRequest_,
+      _OptionGroup_, _sharedSettings_) {
+        $componentController = _$componentController_;
+        $q = _$q_;
+        $rootScope = _$rootScope_;
+        $scope = $rootScope.$new();
+        LeaveRequest = _LeaveRequest_;
+        leaveRequestStatuses = OptionGroupData.getCollection(
+          'hrleaveandabsences_leave_request_status');
+        OptionGroup = _OptionGroup_;
+        sharedSettings = _sharedSettings_;
+        spyOn($scope, '$emit').and.callThrough();
+        spyOn(OptionGroup, 'valuesOf').and.callThrough();
+        spyOn(LeaveRequest, 'all').and.callThrough();
+      }]));
+
+    beforeEach(function () {
+      ctrl = $componentController('leaveWidgetNextLeave', {
+        $scope: $scope
+      });
+      controllerOnChanges.setupController(ctrl);
+    });
+
+    it('should be defined', function () {
+      expect(ctrl).toBeDefined();
+    });
+
+    describe('on init', function () {
+      it('sets balance deduction equal to 0', function () {
+        expect(ctrl.balanceDeduction).toBe(0);
+      });
+
+      it('sets next leave request to NULL', function () {
+        expect(ctrl.nextLeaveRequest).toBe(null);
+      });
+
+      it('sets request status equal to an empty object', function () {
+        expect(ctrl.requestStatus).toEqual({});
+      });
+
+      it('fires a leave widget child is loading event', function () {
+        expect($scope.$emit).toHaveBeenCalledWith(
+          'LeaveWidget::childIsLoading', childComponentName);
+      });
+    });
+
+    describe('bindings and dependencies', function () {
```

Comment:
this section is not needed, you are basically putting the entire test suite in it. just remove it